app/src/main/java/ca/rmen/android/poetassistant/Tts.java

```diff
@@ -192,4 +216,6 @@ public void onUtteranceCompleted(String utteranceId) {
             onUtteranceCompleted();
         }
     }
+
+    private final UtteranceListener mUtteranceListener = new UtteranceListener();
```

Comment:
Since we can init multiple times now (thanks to the public `restart()` method), we should just have one instance of a `UtteranceListener`, not create an anonymous each time we init.